components/UserSettings/DynamicSettingsHeader.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { View, Text, SafeAreaView, StyleSheet, TouchableOpacity, Image } from 'react-native';
+import { useFonts } from 'expo-font';
+import { useNavigation } from 'expo-router';
+
+const DynamicSettingsHeader = ({pageTitle}) => {
+  const navigation = useNavigation();
+
+  let [fontsLoaded] = useFonts({
+    'Poppins-Regular': require('../../assets/fonts/Poppins-Regular.ttf'),
+    'Poppins-Bold': require('../../assets/fonts/Poppins-Bold.ttf'),
+  });
+
+  if (!fontsLoaded) {
+    return null;
+  }
+
+  return (
+    <SafeAreaView style={styles.headerContainer}>
+      <View style={styles.backView}>
+        <TouchableOpacity onPress={() => navigation.goBack()} style={styles.backButton}>
+          <Image source={require('../../assets/images/back-mobile.png')} style={styles.backImage} />
+        </TouchableOpacity>
+      </View>
+
+      <View style={styles.headerView}>
+        <Text numberOfLines={2} style={styles.headerText}>{pageTitle}</Text>
+      </View>
+    </SafeAreaView>
+  );
+};
+
+const styles = StyleSheet.create({

```

Comment:
should be in its own stylign file

Reply:
I haven't done any independent styling files for mobile so far, I honestly prefer it this way.